myapp/forms.py

```diff
@@ -0,0 +1,94 @@
+from django import forms
+from django.contrib.auth import authenticate
+from django.contrib.auth.models import User
+from django.core.validators import MinValueValidator, MaxValueValidator
+
+
+class CheckRequirementsForm(forms.Form):
+    SEX_CHOICES = (
+        ('m', 'male'),
+        ('f', 'female')
+    )
+
+    CEFR = (
+        ("A1", "Beginner"),
+        ("A2", "Elementary"),
+        ("B1", "Intermediate"),
+        ("B2", "Upper-Intermediate"),
+        ("C1", "Advanced"),
+        ("C2", "Proficiency"),
+    )
+    min = MinValueValidator(18)
+    max = MaxValueValidator(80)
+
+    name = forms.CharField(label='name', max_length=100, required=True)
+    sex = forms.ChoiceField(label='sex', choices=SEX_CHOICES, initial='m', widget=forms.RadioSelect, required=True)
+    age = forms.IntegerField(label='age', validators=[min, max], initial=18, required=True)
+    english_level = forms.ChoiceField(label='english level', choices=CEFR)
+
+    def clean(self):
+        cleaned_data = super().clean()
+        sex = cleaned_data.get('sex')
+        age = int(cleaned_data.get('age'))
+        level = cleaned_data.get('english_level')
+
+        is_fit = (sex == 'm' and age >= 20 and level in ['C1', 'C2']) or \
+                 (sex == 'f' and age >= 22 and level in ['B2', 'C1', 'C2'])
+
+        cleaned_data['is_fit'] = is_fit
+        return cleaned_data
```

Comment:
I changed to use seft.is_fit, but it cost rewriting forms __init__  >> added 3 new lines

using add_error, as I understand, doesn't let us throw data to view and get a negative answer in case if a person filled all data correctly, but doesn't fit us by requirements - flow will be stopped at form validation level - or there is any way?

Reply:
Contact me in telegram after lesson and we will discuss all available options

Reply:
According to the conversation, logic has been moved to class CheckRequirementsForm, method clean (with add_error).
__init__ deleted - yes, it works without rewriting